File: baekjoon/11725.cs
using System.Collections;
using System.Collections.Generic;
using static System.Console;
using System.Text;

namespace Project1
{
    public class Class1
    {
        static int N, M;
        static List<int>[] edge = new List<int>[100010];
        static int[] answer = new int[100010];

        static void bfs()
        {
            Queue<int> q = new Queue<int>();           
            q.Enqueue(1);

            while(q.Count != 0)
            {
                int v = (int)q.Dequeue();

                foreach(int x in edge[v])
                {
                    if (answer[x] != 0) continue;
                    answer[x] = v;
                    q.Enqueue(x);
                }
            }
        }

        public static void Main(string[] args)
        {
            int n = int.Parse(ReadLine());
            for (int i = 0; i < 100010; i++) edge[i] = new List<int>();
            string[] s;

            for(int i = 0; i < n-1; i++)
            {
                int a, b;
                s = ReadLine().Split(' ');
                a = int.Parse(s[0]);
                b = int.Parse(s[1]);
                edge[a].Add(b);
                edge[b].Add(a);
            }

            bfs();

            //for (int i = 2; i <= n; i++)
            //    WriteLine(answer[i]);

            //string str = "";
            //for(int i = 2; i <= n; i++)
            //{
            //    str += answer[i] + "\n";
            //}

            StringBuilder sb = new StringBuilder();
            for(int i = 2; i <= n; i++)
            {
                sb.Append(answer[i]).Append("\n");
            }

            Write(sb);
        }

        
    }
}

File: baekjoon/11725.py
import sys
# dequeue가 queue보다 훨씬 빠르다? 
from collections import deque

n = int(input())
edge = [[] for _ in range(n+1)]
answer = [0 for _ in range(n+1)]


def bfs():
    q = deque()
    q.append(1)

    while q:
        v = q.popleft()

        for x in edge[v]:
            if answer[x] != 0: continue
            answer[x] = v
            q.append(x)


for i in range(0, n-1):
    a, b = map(int, sys.stdin.readline().split())
    edge[a].append(b)
    edge[b].append(a)

bfs()

for i in range(2, n+1):
    print(answer[i])


